helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/constraints/MaxCapacityUsageInstanceConstraint.java

```diff
@@ -31,13 +33,17 @@
  * It is a greedy approach since it evaluates only on the most used capacity key.
  */
 class MaxCapacityUsageInstanceConstraint extends UsageSoftConstraint {
+  private static final Logger LOG = LoggerFactory.getLogger(MaxCapacityUsageInstanceConstraint.class.getName());
 
   @Override
   protected double getAssignmentScore(AssignableNode node, AssignableReplica replica,
       ClusterContext clusterContext) {
     float estimatedMaxUtilization = clusterContext.getEstimatedMaxUtilization();
     float projectedHighestUtilization =
-        node.getGeneralProjectedHighestUtilization(replica.getCapacity());
-    return computeUtilizationScore(estimatedMaxUtilization, projectedHighestUtilization);
+            node.getGeneralProjectedHighestUtilization(replica.getCapacity(), clusterContext.getPreferredScoringKey());
+    double utilizationScore = computeUtilizationScore(estimatedMaxUtilization, projectedHighestUtilization);
+    LOG.info("[DEPEND-29018] clusterName: {}, estimatedMaxUtilization: {}, projectedHighestUtilization: {}, utilizationScore: {}, preferredScoringKey: {}",
```

Comment:
Please consider avoiding include user specific ticket here.  We want Helix to be a generic service.

Reply:
+1. Also let's not have this at info level, debug level should be better. Otherwise, it will be per replica per partition per resource log. That's too much.

Reply:
this PR was used to test these changes in ei and get some numbers, will not be merged. I will shortly put up a PR cleaning up the logs.